org.eclipse.winery.repository.ui/src/app/wineryLicenseModule/wineryLicenceFactory.ts

```diff
@@ -0,0 +1,53 @@
+/********************************************************************************
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache Software License 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+import { WineryLicense } from './wineryLicense';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { WineryUrlBasedLicense } from './wineryUrlBasedLicense';
+import { Observable } from 'rxjs/Rx';
+import { forkJoin } from 'rxjs/index';
+
+export class WineryLicenceFactory {
+    private _allLicences: WineryLicense[];
+
+    constructor() {
+        this._allLicences = [
+            new WineryLicense('Apache-2.0', 'assets/licenses/Apache-2.0.txt'),
+            new WineryLicense('EPL-2.0', 'assets/licenses/EPL-2.0.txt'),
+            new WineryLicense('Proprietary', 'assets/licenses/Proprietary.txt')
+        ];
+    }
+
+    loadLicenseTexts(http: HttpClient): Observable<string[]> {
+        // create an array of observables to execute them later on parallel.
+        const allObservables = this._allLicences.map(license => {
+            const headers = new HttpHeaders({ 'Accept': 'text/plain' });
+            return http.get(license.LICENSE_FILE_URL, { headers: headers, responseType: 'text' })
+                .map((result: string) => {
```

Comment:
This is the old way of doing a mapping in rxjs. Please use the version 6: https://stackoverflow.com/a/50235504/6592788.

Please check all your occurrences.